# Vidarbha Voting Percentage Analysis

This project analyses voter turnout in Vidarbha across the 2009, 2014, 2019 and 2024 Lok Sabha elections.

The analysis looks at changes in voting percentage across districts and identifies patterns that could form the basis of a data-driven news story.

In [1]:
import pandas as pd

In [4]:
import os
os.getcwd()

'C:\\Users\\Lenovo\\Downloads'

In [6]:
os.listdir()

['.ipynb_checkpoints',
 '2002 (1).pdf',
 '2002 (2).pdf',
 '2002 (3).pdf',
 '2002 (4).pdf',
 '2002 (5).pdf',
 '2002 (6).pdf',
 '2002 (7).pdf',
 '2002 (8).pdf',
 '2002 (9).pdf',
 '2002.pdf',
 '2026-02-26_Study on the Status of Platform.pdf',
 '2026q1-alphabet-earnings-release.pdf',
 '4gC9Kh42ld.pdf',
 'crops.pdf',
 'data class',
 'desktop.ini',
 'Favorites',
 'gemini-code-1787479225551.md',
 'india_state_renewable_energy_379_rows.csv',
 'NAFISA 10TH MARKSHEET.pdf',
 'NAFISA 12TH MARKSHEET.pdf',
 'NAFISA COVIDE.pdf',
 'NAFISA GRADUATION MARKSHEET.pdf',
 'NAFISA IMAGE.jpeg',
 'NAFISA SIGN.jpeg',
 'Nafisa_Beautifulsoup_homework.ipynb',
 'ojsadmin,+3258-12763-9-ED (1).pdf',
 'ojsadmin,+3258-12763-9-ED.pdf',
 'python-3.13.14-amd64 (1).exe',
 'python-3.13.14-amd64.exe',
 'python-manager-26.3.msix',
 'python-wrangler-setup-0.3.0 (1).exe',
 'session_1',
 'session_1.ipynb',
 'Session_4_BeautifulSoup_follow_along (1).ipynb',
 'Session_4_BeautifulSoup_follow_along (2).ipynb',
 'Session_4_BeautifulS

In [9]:
import pandas as pd

df = pd.read_excel("Vidarbha_Voting_Percentage.xlsx", header=None)

In [10]:
df

,0,1,2,3,4,5,6,7,8,9,10,11
0,NaN,NaN,NaN,Voting Percentage,NaN,NaN,NaN,NaN,12 noon update,NaN,Nagpur district,NaN
1,Centre,Sr No,District Name,2024,2019.0000,2014.0000,2009.0000,NaN,NaN,NaN,Ngp north,0.5805
2,Nagpur,1,Akola,0.6498,0.5836,0.5645,0.5438,NaN,NaN,NaN,Ngp east,0.5870
3,Nagpur,2,Amravati,0.6557,0.6135,0.6485,0.6214,NaN,NaN,NaN,Ngp south,0.5620
4,Nagpur,3,Bhandara,0.6942,0.6855,0.7140,0.6932,NaN,NaN,NaN,Ngp southwest,0.5449
5,Nagpur,4,Gondiya,0.6953,0.6724,0.6949,0.6851,NaN,NaN,NaN,Ngp west,0.5568
6,Nagpur,5,Buldhana,0.7032,0.6555,0.6680,0.6660,NaN,NaN,NaN,Ngp central,0.5501
7,Nagpur,6,Chandrapur,0.7127,0.6562,0.6736,0.6353,NaN,NaN,NaN,urban,0.5636
8,Nagpur,7,Gadchiroli,0.7368,0.7017,0.6703,0.6749,NaN,NaN,NaN,Ramtek,0.7127
9,Nagpur,8,Nagpur,0.6049,0.5808,0.6147,0.5747,NaN,NaN,NaN,Kamptee,0.6305


In [11]:
df.shape

(15, 12)

In [12]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 15 entries, 0 to 14
Data columns (total 12 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   0       12 non-null     str    
 1   1       12 non-null     object 
 2   2       13 non-null     str    
 3   3       14 non-null     object 
 4   4       13 non-null     float64
 5   5       13 non-null     float64
 6   6       13 non-null     float64
 7   7       0 non-null      float64
 8   8       1 non-null      str    
 9   9       0 non-null      float64
 10  10      15 non-null     str    
 11  11      14 non-null     float64
dtypes: float64(6), object(2), str(4)
memory usage: 1.5+ KB


## 4. Cleaning the Dataset

The original Excel sheet contains the main district-level voting data as well as a separate Nagpur voting update.

For this analysis, we focus first on the district-level voting percentages for the 2009, 2014, 2019 and 2024 Lok Sabha elections.

In [13]:
main = df.iloc[1:12, 0:7].copy()

In [14]:
main.columns = [
    "Centre",
    "Sr_No",
    "District",
    "2024",
    "2019",
    "2014",
    "2009"
]

### Cleaned Dataset

The columns are renamed to make the dataset easier to understand and analyse.

In [15]:
main

,Centre,Sr_No,District,2024,2019,2014,2009
1,Centre,Sr No,District Name,2024,2019.0000,2014.0000,2009.0000
2,Nagpur,1,Akola,0.6498,0.5836,0.5645,0.5438
3,Nagpur,2,Amravati,0.6557,0.6135,0.6485,0.6214
4,Nagpur,3,Bhandara,0.6942,0.6855,0.7140,0.6932
5,Nagpur,4,Gondiya,0.6953,0.6724,0.6949,0.6851
6,Nagpur,5,Buldhana,0.7032,0.6555,0.6680,0.6660
7,Nagpur,6,Chandrapur,0.7127,0.6562,0.6736,0.6353
8,Nagpur,7,Gadchiroli,0.7368,0.7017,0.6703,0.6749
9,Nagpur,8,Nagpur,0.6049,0.5808,0.6147,0.5747
10,Nagpur,9,Wardha,0.683,0.6235,0.6671,0.6553


## 5. Converting Voting Figures into Percentages

The voting figures are stored as decimal values. We convert them into percentages so that they are easier to interpret and compare.

In [16]:
for year in ["2024", "2019", "2014", "2009"]:
    main[year] = main[year] * 100

In [17]:
main

,Centre,Sr_No,District,2024,2019,2014,2009
1,Centre,Sr No,District Name,202400,201900.00,201400.00,200900.00
2,Nagpur,1,Akola,64.98,58.36,56.45,54.38
3,Nagpur,2,Amravati,65.57,61.35,64.85,62.14
4,Nagpur,3,Bhandara,69.42,68.55,71.40,69.32
5,Nagpur,4,Gondiya,69.53,67.24,69.49,68.51
6,Nagpur,5,Buldhana,70.32,65.55,66.80,66.60
7,Nagpur,6,Chandrapur,71.27,65.62,67.36,63.53
8,Nagpur,7,Gadchiroli,73.68,70.17,67.03,67.49
9,Nagpur,8,Nagpur,60.49,58.08,61.47,57.47
10,Nagpur,9,Wardha,68.3,62.35,66.71,65.53


## 6. Basic Statistical Analysis

We calculate basic statistics for voting percentages across the four elections. This helps us understand the average, minimum and maximum turnout in the districts covered by the dataset.

In [18]:
main[["2024", "2019", "2014", "2009"]].describe()

,2019,2014,2009
count,11.000000,11.000000,11.000000
mean,18413.106364,18368.920000,18321.740000
std,60855.718143,60704.541864,60554.358502
min,58.080000,56.450000,54.380000
25%,61.850000,65.705000,62.835000
50%,65.620000,66.800000,65.530000
75%,67.895000,68.425000,68.000000
max,201900.000000,201400.000000,200900.000000


### Average Voting Percentage by Election

We calculate the average voting percentage across the districts for each election year to see how turnout has changed over time.

In [19]:
main[["2024", "2019", "2014", "2009"]].mean()

2024    18462.052727
2019    18413.106364
2014        18368.92
2009        18321.74
dtype: object

In [20]:
main[["2024", "2019", "2014", "2009"]].mean()

2024    18462.052727
2019    18413.106364
2014        18368.92
2009        18321.74
dtype: object

## 7. Highest and Lowest Voting Percentages

We identify the districts with the highest and lowest voting percentages in each election year. This helps us understand the differences in voter turnout across districts.

In [21]:
for year in ["2024", "2019", "2014", "2009"]:
    highest = main.loc[main[year].idxmax(), ["District", year]]
    lowest = main.loc[main[year].idxmin(), ["District", year]]
    
    print(year)
    print("Highest:", highest.to_dict())
    print("Lowest:", lowest.to_dict())
    print()

2024
Highest: {'District': 'District Name', '2024': 202400}
Lowest: {'District': 'Nagpur', '2024': 60.49}

2019
Highest: {'District': 'District Name', '2019': 201900.0}
Lowest: {'District': 'Nagpur', '2019': 58.08}

2014
Highest: {'District': 'District Name', '2014': 201400.0}
Lowest: {'District': 'Akola', '2014': 56.45}

2009
Highest: {'District': 'District Name', '2009': 200900.0}
Lowest: {'District': 'Akola', '2009': 54.379999999999995}



## 8. Change in Voting Percentage: 2009 to 2024

To examine long-term changes in voter turnout, we calculate the difference between the voting percentage in 2024 and 2009 for each district.

A positive value indicates an increase in turnout, while a negative value indicates a decline.

In [25]:
main["Change_2009_2024"] = main["2024"] - main["2009"]

### District-wise Change

The table below shows the voting percentage in 2009 and 2024 and the change between the two elections.

In [26]:
main[["District", "2009", "2024", "Change_2009_2024"]]

,District,2009,2024,Change_2009_2024
1,District Name,200900.00,202400,1500.0
2,Akola,54.38,64.98,10.6
3,Amravati,62.14,65.57,3.43
4,Bhandara,69.32,69.42,0.1
5,Gondiya,68.51,69.53,1.02
6,Buldhana,66.60,70.32,3.72
7,Chandrapur,63.53,71.27,7.74
8,Gadchiroli,67.49,73.68,6.19
9,Nagpur,57.47,60.49,3.02
10,Wardha,65.53,68.3,2.77


### District-wise Change

The table below shows the voting percentage in 2009 and 2024 and the change between the two elections.

In [27]:
main[["District", "2009", "2024", "Change_2009_2024"]]

,District,2009,2024,Change_2009_2024
1,District Name,200900.00,202400,1500.0
2,Akola,54.38,64.98,10.6
3,Amravati,62.14,65.57,3.43
4,Bhandara,69.32,69.42,0.1
5,Gondiya,68.51,69.53,1.02
6,Buldhana,66.60,70.32,3.72
7,Chandrapur,63.53,71.27,7.74
8,Gadchiroli,67.49,73.68,6.19
9,Nagpur,57.47,60.49,3.02
10,Wardha,65.53,68.3,2.77


## 9. Comparing Voting Percentages Across Elections

We compare the average voting percentage across the four Lok Sabha elections to identify whether voter turnout has generally increased or decreased over time.

In [28]:
average_turnout = main[["2024", "2019", "2014", "2009"]].mean()

average_turnout

2024    18462.052727
2019    18413.106364
2014        18368.92
2009        18321.74
dtype: object

### Average Voting Percentage

The following table presents the average voting percentage across the districts for each election year.

In [29]:
average_table = pd.DataFrame({
    "Election Year": ["2024", "2019", "2014", "2009"],
    "Average Voting Percentage": [
        average_turnout["2024"],
        average_turnout["2019"],
        average_turnout["2014"],
        average_turnout["2009"]
    ]
})

average_table

,Election Year,Average Voting Percentage
0,2024,18462.052727
1,2019,18413.106364
2,2014,18368.920000
3,2009,18321.740000


### Highest Average Turnout

We identify the election year with the highest average voting percentage among the districts in the dataset.

In [30]:
average_turnout.idxmax(), average_turnout.max()

('2024', 18462.052727272727)

### Lowest Average Turnout

We identify the election year with the lowest average voting percentage among the districts in the dataset.

## 10. Districts With the Largest Changes in Turnout

We compare voting percentages in 2009 and 2024 to identify which districts experienced the largest increases or decreases in voter turnout over this period.

In [31]:
main.loc[
    main["Change_2009_2024"].idxmin(),
    ["District", "2009", "2024", "Change_2009_2024"]
]

District            Bhandara
2009                   69.32
2024                   69.42
Change_2009_2024         0.1
Name: 4, dtype: object

### Checking the Voting Percentage Values

Before calculating averages, we check the voting percentage columns for unusually large or incorrect values. Voting percentages should normally be between 0 and 100.

main[["District", "2024", "2019", "2014", "2009"]]

### Identifying Unusual Values

We check whether any voting percentage is above 100, which would indicate that an incorrect value has been included in the dataset.

In [33]:
main[["District", "2024", "2019", "2014", "2009"]].loc[
    (main[["2024", "2019", "2014", "2009"]] > 100).any(axis=1)
]

,District,2024,2019,2014,2009
1,District Name,202400,201900.0,201400.0,200900.0


In [34]:
main = df.iloc[1:12, 0:7].copy()

In [35]:
main = df.iloc[2:13, 0:7].copy()

### Correcting the Dataset Structure

The original sheet contains a separate header row within the data area. We remove this row from the analysis and retain only the district-level observations.

In [36]:
main.columns = [
    "Centre",
    "Sr_No",
    "District",
    "2024",
    "2019",
    "2014",
    "2009"
]

### Checking the Cleaned Dataset

We now check the cleaned dataset to make sure that only actual district-level observations are included.

In [37]:
main

,Centre,Sr_No,District,2024,2019,2014,2009
2,Nagpur,1,Akola,0.6498,0.5836,0.5645,0.5438
3,Nagpur,2,Amravati,0.6557,0.6135,0.6485,0.6214
4,Nagpur,3,Bhandara,0.6942,0.6855,0.7140,0.6932
5,Nagpur,4,Gondiya,0.6953,0.6724,0.6949,0.6851
6,Nagpur,5,Buldhana,0.7032,0.6555,0.6680,0.6660
7,Nagpur,6,Chandrapur,0.7127,0.6562,0.6736,0.6353
8,Nagpur,7,Gadchiroli,0.7368,0.7017,0.6703,0.6749
9,Nagpur,8,Nagpur,0.6049,0.5808,0.6147,0.5747
10,Nagpur,9,Wardha,0.683,0.6235,0.6671,0.6553
11,Nagpur,10,Yavatmal,0.6902,0.6690,0.6656,0.6417


In [38]:
main


,Centre,Sr_No,District,2024,2019,2014,2009
2,Nagpur,1,Akola,0.6498,0.5836,0.5645,0.5438
3,Nagpur,2,Amravati,0.6557,0.6135,0.6485,0.6214
4,Nagpur,3,Bhandara,0.6942,0.6855,0.7140,0.6932
5,Nagpur,4,Gondiya,0.6953,0.6724,0.6949,0.6851
6,Nagpur,5,Buldhana,0.7032,0.6555,0.6680,0.6660
7,Nagpur,6,Chandrapur,0.7127,0.6562,0.6736,0.6353
8,Nagpur,7,Gadchiroli,0.7368,0.7017,0.6703,0.6749
9,Nagpur,8,Nagpur,0.6049,0.5808,0.6147,0.5747
10,Nagpur,9,Wardha,0.683,0.6235,0.6671,0.6553
11,Nagpur,10,Yavatmal,0.6902,0.6690,0.6656,0.6417


### Checking for Invalid Voting Percentages

Since voting percentages should fall between 0 and 100, we check again for any values above 100.

In [39]:
main[["District", "2024", "2019", "2014", "2009"]].loc[
    (main[["2024", "2019", "2014", "2009"]] > 100).any(axis=1)
]

,District,2024,2019,2014,2009


## 6. Basic Statistical Analysis

After cleaning the dataset, we calculate the average voting percentage for each election year. This allows us to compare overall voter turnout across the four elections.

In [40]:
average_turnout = main[["2024", "2019", "2014", "2009"]].mean()

average_turnout

2024    0.680536
2019    0.642418
2014    0.654118
2009    0.636373
dtype: object

In [41]:
for year in ["2024", "2019", "2014", "2009"]:
    main[year] = main[year] * 100

In [42]:
average_turnout

2024    0.680536
2019    0.642418
2014    0.654118
2009    0.636373
dtype: object

### Average Voting Percentage

The voting figures have now been converted from decimal values into percentages. We recalculate the average turnout for each election year using the cleaned data.

In [44]:
average_turnout = main[["2024", "2019", "2014", "2009"]].mean()

average_turnout

2024    68.053636
2019    64.241818
2014    65.411818
2009    63.637273
dtype: object

### Election With the Highest Average Turnout

We identify the election year with the highest average voting percentage across the districts in the dataset.

In [45]:
average_turnout.idxmax(), average_turnout.max()

('2024', 68.05363636363636)

### Election With the Lowest Average Turnout

We identify the election year with the lowest average voting percentage across the districts in the dataset.

In [46]:
average_turnout.idxmin(), average_turnout.min()

('2009', 63.63727272727272)

## 7. Comparing Average Turnout Across Elections

The table below summarises the average voting percentage across the four Lok Sabha elections covered by the dataset. This allows us to observe the overall direction of voter turnout over time.

In [47]:
average_table = pd.DataFrame({
    "Election Year": ["2009", "2014", "2019", "2024"],
    "Average Voting Percentage": [
        average_turnout["2009"],
        average_turnout["2014"],
        average_turnout["2019"],
        average_turnout["2024"]
    ]
})

average_table

,Election Year,Average Voting Percentage
0,2009,63.637273
1,2014,65.411818
2,2019,64.241818
3,2024,68.053636


In [48]:
average_table = pd.DataFrame({
    "Election Year": ["2009", "2014", "2019", "2024"],
    "Average Voting Percentage": [
        average_turnout["2009"],
        average_turnout["2014"],
        average_turnout["2019"],
        average_turnout["2024"]
    ]
})

average_table

,Election Year,Average Voting Percentage
0,2009,63.637273
1,2014,65.411818
2,2019,64.241818
3,2024,68.053636


## 8. Change in Voting Percentage Between 2009 and 2024

To examine the long-term change in voter turnout, we calculate the difference between each district's voting percentage in 2009 and 2024.

A positive value means voting percentage increased, while a negative value means it decreased.

main["Change_2009_2024"] = main["2024"] - main["2009"]

main[["District", "2009", "2024", "Change_2009_2024"]]

### Ranking Districts by Change

We rank the districts according to the change in voting percentage between 2009 and 2024. This helps identify the districts where turnout increased or decreased the most.

In [50]:
main[["District", "2009", "2024", "Change_2009_2024"]].sort_values(
    "Change_2009_2024",
    ascending=False
)

,District,2009,2024,Change_2009_2024
2,Akola,54.38,64.98,10.6
7,Chandrapur,63.53,71.27,7.74
8,Gadchiroli,67.49,73.68,6.19
12,Washim,60.87,66.01,5.14
11,Yavatmal,64.17,69.02,4.85
6,Buldhana,66.60,70.32,3.72
3,Amravati,62.14,65.57,3.43
9,Nagpur,57.47,60.49,3.02
10,Wardha,65.53,68.3,2.77
5,Gondiya,68.51,69.53,1.02


### District With the Largest Increase

We identify the district that recorded the largest increase in voting percentage between 2009 and 2024.

main.loc[
    main["Change_2009_2024"].idxmax(),
    ["District", "2009", "2024", "Change_2009_2024"]
]

### District With the Largest Decline

We identify the district that recorded the largest decline in voting percentage between 2009 and 2024.

In [52]:
main.loc[
    main["Change_2009_2024"].idxmin(),
    ["District", "2009", "2024", "Change_2009_2024"]
]

District            Bhandara
2009                   69.32
2024                   69.42
Change_2009_2024         0.1
Name: 4, dtype: object

## 9. Districts With Increased or Decreased Turnout

We count how many districts recorded an increase and how many recorded a decrease in voting percentage between 2009 and 2024.

In [53]:
increased = (main["Change_2009_2024"] > 0).sum()
decreased = (main["Change_2009_2024"] < 0).sum()
unchanged = (main["Change_2009_2024"] == 0).sum()

print("Districts with increased turnout:", increased)
print("Districts with decreased turnout:", decreased)
print("Districts with unchanged turnout:", unchanged)

Districts with increased turnout: 11
Districts with decreased turnout: 0
Districts with unchanged turnout: 0


## 10. Change in Voting Percentage Between 2019 and 2024

We compare voter turnout in the most recent two elections to determine whether the increase seen in 2024 was widespread across the districts.

In [54]:
main["Change_2019_2024"] = main["2024"] - main["2019"]

main[["District", "2019", "2024", "Change_2019_2024"]]

,District,2019,2024,Change_2019_2024
2,Akola,58.36,64.98,6.62
3,Amravati,61.35,65.57,4.22
4,Bhandara,68.55,69.42,0.87
5,Gondiya,67.24,69.53,2.29
6,Buldhana,65.55,70.32,4.77
7,Chandrapur,65.62,71.27,5.65
8,Gadchiroli,70.17,73.68,3.51
9,Nagpur,58.08,60.49,2.41
10,Wardha,62.35,68.3,5.95
11,Yavatmal,66.90,69.02,2.12


### Districts With Higher Turnout in 2024

We count the number of districts where voter turnout increased between 2019 and 2024.

In [55]:
(main["Change_2019_2024"] > 0).sum()

np.int64(11)

### District With the Largest Increase From 2019 to 2024

We identify the district with the largest increase in voting percentage between the two most recent elections.

In [56]:
main.loc[
    main["Change_2019_2024"].idxmax(),
    ["District", "2019", "2024", "Change_2019_2024"]
]

District            Akola
2019                58.36
2024                64.98
Change_2019_2024     6.62
Name: 2, dtype: object

## 11. Overall Change in Average Turnout

We compare the average voting percentage in 2009 with the average voting percentage in 2024 to measure the overall change in voter turnout across the districts.

In [57]:
overall_change = average_turnout["2024"] - average_turnout["2009"]

overall_change

4.416363636363634

# 12. Key Findings

The analysis shows that voter turnout in the districts covered by the dataset increased over the long term.

- The average voting percentage increased from approximately 63.64% in 2009 to 68.05% in 2024.
- This represents an increase of approximately 4.42 percentage points.
- All 11 districts recorded higher turnout in 2024 than in 2009.
- All 11 districts also recorded higher turnout in 2024 than in 2019.
- Akola recorded the largest increase between 2019 and 2024, rising from 58.36% to 64.98%, an increase of 6.62 percentage points.

These findings suggest that the rise in voter turnout in 2024 was widespread across the districts in the dataset rather than being limited to a small number of districts.

# 13. Story Pitch

## Vidarbha Sees Broad Rise in Voter Turnout

Voter turnout across the Vidarbha districts covered in the dataset increased in the 2024 Lok Sabha election, with all 11 districts recording higher participation than in 2009 and 2019. Average turnout rose from 63.64% in 2009 to 68.05% in 2024, an increase of 4.42 percentage points. Akola recorded the biggest increase between 2019 and 2024, with turnout rising by 6.62 percentage points to 64.98%.

The data raises a broader reporting question: what drove the increase in voter participation across Vidarbha in 2024? A follow-up story could examine whether the rise was linked to voter-awareness campaigns, changes in the electorate, local political issues, or differences in urban and rural participation. Interviews with election officials, political analysts and voters could help explain the reasons behind the trend.

# 13. Story Pitch

## Vidarbha Sees Broad Rise in Voter Turnout

Voter turnout across the Vidarbha districts covered in the dataset increased in the 2024 Lok Sabha election, with all 11 districts recording higher participation than in 2009 and 2019. Average turnout rose from 63.64% in 2009 to 68.05% in 2024, an increase of 4.42 percentage points. Akola recorded the biggest increase between 2019 and 2024, with turnout rising by 6.62 percentage points to 64.98%.

The data raises a broader reporting question: what drove the increase in voter participation across Vidarbha in 2024? A follow-up story could examine whether the rise was linked to voter-awareness campaigns, changes in the electorate, local political issues, or differences in urban and rural participation. Interviews with election officials, political analysts and voters could help explain the reasons behind the trend.